creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -0,0 +1,504 @@
+        """
+        Create study, sequencing-center in Data Service
+        """
+        self.logger.info(
+            f"Initializing study {self.study_id} in "
+            f"Data Service {DATASERVICE_URL}"
+        )
+        payloads = [
+            {
+                "kf_id": self.study_id,
+                "name": self.study_id,
+                "external_id": self.study_id,
+                "endpoint": "studies"
+            },
+        ]
+        for sc in self.sequencing_centers:
+            payloads.append(
+                {
+                    "kf_id": sc["kf_id"],
+                    "name": sc["name"],
+                    "external_id": sc["name"],
+                    "endpoint": "sequencing-centers"
+                }
+            )
+        for p in payloads:
+            endpoint = p.pop("endpoint", None)
+            url = f"{DATASERVICE_URL}/{endpoint}"
+            self.logger.info(f"Creating {url}")
+            try:
+                resp = requests.post(url, json=p)
+                resp.raise_for_status()
+            except requests.exceptions.HTTPError:
+                ok = resp.status_code == 400 and "already exists" in resp.text
+                if not ok:
+                    raise
+
+    def _kf_id_generator(self, prefix):
+        """
+        Generator Data Service KF IDS
+
+        Ex:
+        'PT_0004PEDE'
+        'SA_D167JSHP'
+        'DM_ZZZZZZZZ'
+        'ST_00000000'
+        """
+        assert len(prefix) == 2, 'Prefix must be two characters'
+        prefix = prefix.upper()
+
+        def generator():
+            return '{0}_{1:0>8}'.format(
+                prefix, b32.encode(ra.randint(0, 32**8-1))
+            )
+
+        return generator
+
+    def _create_bio_manifest(self, create=False):
+        """
+        Create a tabular data file representing a Biospecimen manifest. This
+        just includes the necessary attributes to ingest Biospecimen entities.
+        N rows are generated, one for each Biospecimen. The number needed is
+        up to the user. The simplest solution is to have one per participant.
+        """
+        verb = "Creating" if create else "Reading"
+        n = self.total_specimens
+        self.logger.info(
+            f"{verb} bio manifest with {n} participants, "
+            "each with 1 specimen"
+        )
+        fp = os.path.join(self.data_dir, 'bio_manifest.tsv')
+        if not create:
+            return read_df(fp)
+
+        _range = range(n)
+        bio_dict = {
+            "sample_id": [f"SM-{i}" for i in _range],
+            "participant_id": [f"CARE-{i}" for i in _range],
+            "gender": [ra.choice(("Female", "Male")) for _ in _range],
+            "volume": [100] * n,
+            "concentration": [30] * n,
+            "family_id": [ra.choice(("FA-1", "FA-2")) for _ in _range],
+            "tissue_type": [ra.choice(("Blood", "Saliva")) for _ in _range],
+        }
+        id_gen = self._kf_id_generator
```

Comment:
This method is the same as before except that I added the logic to inject the fixed kf ids